l10n_id_taxform_period/models/tax_period.py

```diff
@@ -0,0 +1,154 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Prime Force Indonesia
+# Copyright 2016 OpenSynergy Indonesia
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import models, fields, api
+from openerp.tools.translate import _
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+
+
+class TaxYear(models.Model):
+    _name = "l10n_id.tax_year"
+    _description = "Tax Year"
+    _order = "date_start asc, id"
+
+    name = fields.Char(
+        string="Tax Year",
+        required=True,
+    )
+    code = fields.Char(
+        string="Code",
+        required=True,
+    )
+    date_start = fields.Date(
+        string="Date Start",
+        required=True,
+    )
+    date_end = fields.Date(
+        string="Date End",
+        required=True,
+    )
+    period_ids = fields.One2many(
+        string="Periods",
+        comodel_name="l10n_id.tax_period",
+        inverse_name="year_id",
+    )
+
+    @api.constrains("date_start", "date_end")
+    def _check_range(self):
+        if self.date_end <= self.date_start:
+            strWarning = _("The start date must precede it's end date")
+            raise models.ValidationError(strWarning)
```

Comment:
A transitive import is not a good idea because you use more paths to get the main library.
This is similar to have: `from openerp.addons.other_module.models.model import ValidationError`
Or have:
```python
# file1.py
import os

#file2.py
from file1 import os  # Why not use directly "import os"?
```

Using transitive import requires more caution for new versions.
A example is the deprecated [except_orm](https://github.com/odoo/odoo/blob/dd28f9bf78c35c878eaf35622212e5a40f90fa40/openerp/exceptions.py#L21).

The version of models.py before of deprecating:
```python
from .exceptions import except_osv
...
```

The version of models.py after deprecating
```python
from .exceptions import UserError, RedirectWarning
...
```


You old `mymodule.py` with transitive import will be fail `from openerp.models import except_osv` because now is not directly imported from transitive `models.py` file. But if we have a `from openerp.exceptions import except_osv` works perfectly.

We have a similar deprecating case for `exception.warning`  from 8.0 and 9.0 and the lint check this case.
And we have a guideline: https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#idioms
see `Exceptions: Use from openerp.exceptions import Warning as UserError (v8) or from openerp.exceptions import UserError (v9) or find a more appropriate exception in openerp.exceptions.py`

And the lint check this sentences with `openerp.exceptions` but not is checked with `models.exceptions`